homework/src/index.js

```diff
@@ -30,17 +33,115 @@
+    const table = createAndAppend('table', container);
+    const tbody = createAndAppend('tbody', table);
+    const tr1 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr1, {
+      text: 'Repository: ',
+    });
+    const td2 = createAndAppend('td', tr1);
+    createAndAppend('a', td2, {
+      href: repository.html_url,
+      target: '_blank',
+      text: repository.name
+    });
+    const tr2 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr2, {
+      text: 'Description:'
+    });
+    createAndAppend('td', tr2, {
+      text: repository.description
+    });
+    const tr3 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr3, {
+      text: 'Forks:'
+    });
+    createAndAppend('td', tr3, {
+      text: repository.forks
+    });
+    const tr4 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr4, {
+      text: 'Updated:'
+    });
+    createAndAppend('td', tr4, {
+      text: repository.updated_at
+    });
+  }
+  function renderContributors(contributors, container) {
+    container.innerHTML = '';
+    const h2 = createAndAppend('h2', container, {
+      text: 'Contributors'
+    });
+    const ul = createAndAppend('ul', container);
+
+    contributors.forEach(contributor => {
+      const li = createAndAppend('li', ul);
+      createAndAppend('a', li, {
+        href: contributor.html_url,
+        target: '_blank',
+        text: contributor.login
+      });
+      createAndAppend('img', li, {
+        src: contributor.avatar_url
+      });
+      createAndAppend('div', li, {
+        text: contributor.contributions
+      });
     });
   }
 
+  async function main(url) {
+    const root = document.getElementById('root');
+    try {
+      const repositories = await fetchJSON(url);
+
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
+      console.log(repositories);
+      const header = createAndAppend('header', root);
+      const h1 = createAndAppend('h2', header, {
+        text: 'HYF Repositories'
+      });
+      const select = createAndAppend('select', header);
+      const leftHandContainer = createAndAppend('div', root, {
+        id: 'left-hand'
+      });
+      const rightHandContainer = createAndAppend('div', root, {
+        id: 'right-hand'
+      });
+      repositories.forEach((repository, index) => {
+        createAndAppend('option', select, {
+          text: repository.name,
+          value: index
+        });
+      });
+      select.addEventListener('change', async () => {
+        const index = select.value;
+        const repository = repositories[index];
+        renderRepository(repository, leftHandContainer);
+        try {
+          const contributors = await fetchJSON(repository.contributors_url);
+          renderContributors(contributors, rightHandContainer);
+        }
+        catch (err) {
+          err.message;
+        }
+      });
+      renderRepository(repositories[0], leftHandContainer);
+      try {
```

Comment:
Here you are nesting a `try` block inside another `try` block (the one that starts in line 97). Since you are already in a `try` block here there is no reason to start another one as the outer `try` block will already catch errors. In this case it is best to remove the inner `try` block (remove lines 132 and 135-137).

Note that you do need the try/catch block in the event handler function that starts in line 119, as this event handler is executed asynchronously, that is after function `main()` has already finished executing.